lib/ansible/plugins/action/gather_facts.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import time
+
+from ansible.module_utils.common._collections_compat import MutableMapping
+
+from ansible import constants as C
+from ansible.plugins.action import ActionBase
+from ansible.utils.vars import combine_vars
+
+
+class ActionModule(ActionBase):
+
+    def _get_module_args(self, fact_module, task_vars):
+        mod_args = task_vars.get('ansible_facts_modules', {}).get(fact_module, {})
```

Comment:
Should this technically also go through the config path, passing task_vars as `variables` instead of directly to task_vars?